sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java

```diff
@@ -404,9 +406,11 @@ protected ReactorReceiver createConsumer(String entityPath, Receiver receiver,
         Map<Symbol, Object> linkProperties, boolean requiresAuthorization) {
 
         if (isDisposed()) {
-            return Mono.error(logger.logExceptionAsError(new IllegalStateException(String.format(
```

Comment:
If a connection has been built and the receiver/sender has been working and then unplug the network, this exception will be hit so a non-retryable `IllegalStateException` is thrown. I changed it to a transient AmqpException. This also applies to other similar cases.